DxDiagram new API (#1112)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md

---
---
##### shortDescription
Specifies a custom template for shapes in the toolbox.

##### param(container): dxSVGElement
#include common-ref-elementparam with { element: "shape" }

##### param(data): Object
Information about the currently processed shape.

##### field(data.item): dxDiagramShape
Information about the currently processed shape.

---
The template content must be presented as SVG elements.

> Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template of an individual shape in the toolbox.

Reviewer: Can you clarify what you mean by "a template of an individual shape"?

Author: Diagram has two options for a shape template in the toolbox: one specifies a template for all shapes (diagram.customShapeToolboxTemplate). Another specifies a template for a particular shape (customShape.toolboxTemplate), so users can create a template for every shape.

Reviewer: ```suggestion
Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template for each shape in the toolbox.
```
